imap_processing/spice/geometry.py

```diff
@@ -1 +1,128 @@
-"""Functions for computing geometry using SPICE."""
+"""Functions for computing geometry, many of which use SPICE."""
+
+from collections.abc import Iterable
+from enum import Enum
+from typing import NamedTuple, Optional, Union
+
+import numpy as np
+import spiceypy as spice
+
+from imap_processing.spice.kernels import ensure_spice
+
+
+class SpiceId(NamedTuple):
+    """Class that represents a unique identifier in the NAIF SPICE library."""
+
+    strid: str
+    numid: int
```

Comment:
I think we can get rid of this NamedTuple and just use your Enums directly below. Maybe use IntEnum instead. But then you'd just have `SpiceBody.IMAP.name` which would still return the string IMAP. (The one difference you have is SSB which I think could just be written out in the long form)
```suggestion
```

Reply:
The only issue I see with that is that some SPICE names have spaces in them.

Reply:
Oh interesting... and unfortunate. It would be much cleaner as a normal Enum if we could somehow manage that.